pkgs/development/python-modules/bootstrap/build/default.nix

```
@@ -40,10 +43,24 @@ let
   bootstrap-tomli = buildBootstrapPythonModule tomli {};
 in
 buildBootstrapPythonModule build {
-  propagatedBuildInputs = [
-    bootstrap-packaging
-    bootstrap-pyproject-hooks
-  ] ++ lib.optionals (python.pythonOlder "3.11") [
-    bootstrap-tomli
-  ];
+  # like the installPhase above, but wrapping the pyproject-build command
+  # to set up PYTHONPATH with the correct dependencies.
+  # This allows using `pyproject-build` without propagating its dependencies
+  # into the build environment, which is necessary to prevent
+  # pythonCatchConflicts from raising false positive alerts.
+  # This would happen whenever the package to build has a dependency on
+  # another version of a package that is also a dependency of pyproject-build.
```

```suggestion
  # like the installPhase above, but wrapping the pyproject-build command to set up PYTHONPATH with the correct dependencies.
  # This allows using `pyproject-build` without propagating its dependencies into the build environment,
  # which is necessary to prevent pythonCatchConflicts from raising false positive alerts.
  # This would happen whenever the package to build has a dependency on another version of a package that is also a dependency of pyproject-build.
```
one sentence per line is way easier to read for some people like me